### PR TITLE
iOS pipeline template with Fastlane

### PR DESCRIPTION
**Pull request by olyism**
Creating a separate template for iOS app using Fastlane.

Not all users may want to integrate Fastlane, so I think it's good to keep the rudimentary Swift template as well.

### CODE REVIEW

**Comment by olyism** [comment]
iOS pipeline template is ready for review.

I have minimal experience with developing iOS apps, seeking feedback and help on:
- Cache plugin configuration
  - Is bundler the best approach?
  - Is it caching correctly?
- Fastfile config example (and if including this is helpful)

File: fastlane-ios/.gitignore [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#
+# It is recommended to not store the screenshots in the git repo.
+# Instead, use fastlane to re-generate the screenshots whenever they are needed.
+# For more information about the recommended setup visit:
+# https://docs.fastlane.tools/best-practices/source-control/#source-control
+
+fastlane/report.xml
+fastlane/Preview.html
+fastlane/screenshots/**/*.png
+fastlane/test_output
```
Comment (tommeier) [inline fastlane-ios/.gitignore:83]:
```suggestion
fastlane/test_output
fastlane/secure/*
fastlane/dsyms
```

I've added above in past too, I think it was from Gym (auto building in fastlane) and Auth for secure, but simply following their doc example that you've linked is probobly best

File: fastlane-ios/fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+default_platform(:ios)
+
+platform :ios do
+  
+  project_path = "myProject/myProject.xcodeproj"
+
+  desc "Run iOS tests"
+  lane :test do
+    scan(
```
Comment (tommeier) [inline fastlane-ios/fastlane/Fastfile:24]:
```suggestion
    run_tests(
```

Scan is now the older/original way and an alias of `run_tests` https://docs.fastlane.tools/actions/run_tests/

File: fastlane-ios/fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      project: project_path
+    )
+  end
+
+  desc "Run SwiftLint"
+  lane :lint do
+    sh "swiftlint"
```
Comment (tommeier) [inline fastlane-ios/fastlane/Fastfile:31]:
```suggestion
    swiftlint(mode: :lint)
```
Swiftlint is now core: https://docs.fastlane.tools/actions/swiftlint/

File: fastlane-ios/fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  desc "Run build"
+  lane :build do
+    gym(
```
Comment (tommeier) [inline fastlane-ios/fastlane/Fastfile:36]: Might be too opinionated for here but I've always needed `clean_build_artifacts` & `clear_derived_data` for consistent build outcomes prior to building (even with `clean: true`)

**Review by tommeier — COMMENTED** [review]
Looks good! Its hard to suggest a minimum as ios testing is always _quirky_ and requires a bunch of things, I haven't spun this up so if its all working 👏  Added some minor comments for things I've noticed in the past

File: fastlane-ios/pipeline.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  - label: ":fastlane: Test"
+    command: "bundle exec fastlane ios test"
+    depends_on: "deps"
+    plugins:
+      - cache#v0.6.0:
+          manifest: Gemfile.lock
+          path: vendor
+          restore: file
+    env:
```
Comment (tommeier) [inline fastlane-ios/pipeline.yaml:20]: Could be a good example point for artifact paths:

```
artifact_paths:
    - fastlane/test_output/*
``

**Comment by tommeier** [comment]
Fastfile config example is definitely worth it - will save a lot of time for peeps 👍  
I'm unsure about the cache plugin sorry (I haven't used it before)

**Review by tommeier — APPROVED** [review]
Looks great 👏

**Pull request merged**